### Import Libraries

In [2]:
import pandas as pd
import numpy as np
import os

### Import Data

In [3]:
df_prods = pd.read_csv(r"C:\Users\SigmaEd\Documents\CareerFoundry\Instacart Basket Analysis\02 Data\Prepared Data\products_wrangled.csv")
df_ords = pd.read_csv(r"C:\Users\SigmaEd\Documents\CareerFoundry\Instacart Basket Analysis\02 Data\Prepared Data\orders_wrangled.csv")

### Question 2. Does anything about the data looks off or should be investigated further

In [4]:
df_prods.describe()

Unnamed: 0    product_id      aisle_id  department_id        prices
count  49672.000000  49672.000000  49672.000000   49672.000000  49672.000000
mean   24852.005053  24850.349775     67.762442      11.728942      9.993282
std    14342.265579  14340.705287     38.315784       5.850779    453.615536
min        0.000000      1.000000      1.000000       1.000000      1.000000
25%    12432.750000  12432.750000     35.000000       7.000000      4.100000
50%    24851.500000  24850.500000     69.000000      13.000000      7.100000
75%    37272.250000  37268.250000    100.000000      17.000000     11.100000
max    49692.000000  49688.000000    134.000000      21.000000  99999.000000

In [5]:
df_prods.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49672 entries, 0 to 49671
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Unnamed: 0     49672 non-null  int64  
 1   product_id     49672 non-null  int64  
 2   product_name   49672 non-null  object 
 3   aisle_id       49672 non-null  int64  
 4   department_id  49672 non-null  int64  
 5   prices         49672 non-null  float64
dtypes: float64(1), int64(4), object(1)
memory usage: 2.3+ MB


In [6]:
df_prods

Unnamed: 0  product_id  \
0               0           1   
1               1           2   
2               2           3   
3               3           4   
4               4           5   
...           ...         ...   
49667       49688       49684   
49668       49689       49685   
49669       49690       49686   
49670       49691       49687   
49671       49692       49688   

                                            product_name  aisle_id  \
0                             Chocolate Sandwich Cookies        61   
1                                       All-Seasons Salt       104   
2                   Robust Golden Unsweetened Oolong Tea        94   
3      Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4                              Green Chile Anytime Sauce         5   
...                                                  ...       ...   
49667          Vodka, Triple Distilled, Twist of Vanilla       124   
49668                 En Croute Roast Hazelnut Cranberry        42   
49669                                   Artisan Baguette       112   
49670         Smartblend Healthy Metabolism Dry Cat Food        41   
49671                             Fresh Foaming Cleanser        73   

       department_id  prices  
0                 19     5.8  
1                 13     9.3  
2                  7     4.5  
3                  1    10.5  
4                 13     4.3  
...              ...     ...  
49667              5     5.3  
49668              1     3.1  
49669              3     7.8  
49670              8     4.7  
49671             11    13.5  

[49672 rows x 6 columns]

### Question 2 Answer = max value for prices (99999) seems very high when compared to the mean (9.99)and median (7.10)

### Question 3. Check for mixed-type data in your df_ords dataframe

In [7]:
for col in df_ords.columns.tolist():
  weird = (df_ords[[col]].applymap(type) != df_ords[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_ords[weird]) > 0:
    print (col)

### Question 4. Answer = No mixed-type data found

### Question 5. Check for missing values in your df_ords dataframe

In [8]:
df_ords.isnull().sum()

Unnamed: 0                     0
order_id                       0
user_id                        0
order_number                   0
order_day_or_week              0
order_hour_of_day              0
days_since_prior_order    206209
dtype: int64

In [9]:
df_nan = df_ords[df_ords['days_since_prior_order'].isnull()==True]
df_nan

Unnamed: 0  order_id  user_id  order_number  order_day_or_week  \
0                 0   2539329        1             1                  2   
11               11   2168274        2             1                  2   
26               26   1374495        3             1                  1   
39               39   3343014        4             1                  6   
45               45   2717275        5             1                  3   
...             ...       ...      ...           ...                ...   
3420930     3420930    969311   206205             1                  4   
3420934     3420934   3189322   206206             1                  3   
3421002     3421002   2166133   206207             1                  6   
3421019     3421019   2227043   206208             1                  1   
3421069     3421069   3154581   206209             1                  3   

         order_hour_of_day  days_since_prior_order  
0                        8                     NaN  
11                      11                     NaN  
26                      14                     NaN  
39                      11                     NaN  
45                      12                     NaN  
...                    ...                     ...  
3420930                 12                     NaN  
3420934                 18                     NaN  
3421002                 19                     NaN  
3421019                 15                     NaN  
3421069                 11                     NaN  

[206209 rows x 7 columns]

### Question 5 Answer = 206209 missing values because there is no prior purchase date for a customers first purchase.

### Question 6. Answer = There is no way to fill in missing values here

### Question 7. Check for duplicate values in your df_ords data

In [10]:
df_dups = df_ords[df_ords.duplicated()]
df_dups

Empty DataFrame
Columns: [Unnamed: 0, order_id, user_id, order_number, order_day_or_week, order_hour_of_day, days_since_prior_order]
Index: []

### Question 7 & 8. Answer = There are no duplicate values

### Export to csv

In [11]:
df_ords.to_csv(r"C:\Users\SigmaEd\Documents\CareerFoundry\Instacart Basket Analysis\02 Data\Prepared Data\orders_wrangled1.csv")